src/main.ts

```diff
@@ -170,6 +174,20 @@ export class Purchases {
     }
     this.backend = new Backend(this._API_KEY, httpConfig);
     this.purchaseOperationHelper = new PurchaseOperationHelper(this.backend);
+    this.initialize();
+  }
+
+  private initialize() {
+    this.backend
+      .getBrandingInfo()
```

Comment:
Note this will cause a big increase in requests to this endpoint

Reply:
Ouch, the `initialize()` method is sync.

Using `.getBrandingInfo().then` inside it might cause some weird scenario in which the `purchase` method is called before `.getBrandingInfo()` is loaded. It's rare but it might happen especially in cases of low/slow connectivity.

Maybe we should define an explicit method to preload the branding async or make a breaking change in which we convert `initialize` to async.

Wdyt?

Reply:
Note that `initialize` is private and is currently called in the `constructor`. I guess we could make `configure` (which is the one that calls this) async... I just worry about delaying initialization for a secondary task like getting branding info, since I would assume devs might call the `configure` method very early in their app.

I was thinking that the worst thing that would happen would be that we fetch branding info multiple times right? I don't think that would be a big deal, specially since it shouldn't cause a race condition since the response should be the same... Alternatively, we could write some sort of cancellation or hook to the existing request if any so we only do it once but I feel it's overkill. Wdyt?

Reply:
I would personally prefer to convert `.configure` in async if possible.
Working with hooks and such will make it really dirty compared to the built in solution from the language. And we will then have to maintain that "custom" implementation going forward.

Regarding the fact that branding is a secondary task, I think that now we are having just the branding but later we might have some more stuff to download so moving to `async` now might make our life easier later.

It would be a breaking change, but we could introduce it together with the customisation release (when it will come) to sweeten the change.

For every developer that calls it very early in the app (and doesn't really care about the result) they would just not implement the `.then()` which will have the same behaviour you just proposed. It will eventually load and it might cause a race condition in rare scenarios.

But at least we keep it clean and we have a clearly documented change that states that devs should consider `.configure` async from now on.

So worst case it works as well as in the solution you proposed, but it gives us freedom to load whatever we want in the `.configure` going forward. We are still in beta, some breaking changes will happen.

Reply:
Hmm I'm not so worried about the breaking change but about causing the SDK to be harder to use... Right now the `configure` call will fill the static `Purchases.sharedInstance`, which would be delayed by this change... If someone doesn't want to wait for the SDK to initialize and then try to call the SDK using that static accessor, it might still not be configured. Even if they don't use the static accessor (and they don't want to wait for the `configure` to finish) and want to use the return value of the `configure` being now a `Promise<Purchases>`, they would need to store the value somewhere else that they can check on whether the SDK has already been configured leading to several possible issues IMO.

> later we might have some more stuff to download

That's a valid concern, but for what is worth, most of our mobile SDKs methods are async, but the `configure` method is synchronous. We do prefetch a bunch of information during the `configure` in the background. To solve the issue of devs asking for information before this is done, we currently have some deduping logic for requests, which we might be able to add to purchases-js.

Reply:
What about we make this a topic for the next grooming session?

We can have a chat tomorrow morning or Monday and prepare it to start working on eventual changes on the next cycle (starting Monday).

I created this ticket as a reference for the grooming:
https://linear.app/revenuecat/issue/ECO-1396/[rcbilling]-[sdk]-preloading-branding-and-configuration-at-configure

Reply:
Agreed! This PR is not urgent, so makes sense to talk more about it. Thanks!